core/src/test/java/io/confluent/kafka/schemaregistry/rest/filters/ContextFilterTest.java

```diff
@@ -263,4 +263,51 @@ public void testInvalidContext() {
     );
   }
 
+  @Test
+  public void testSubjectNamedFoo() {
+    String path = "/subjects/foo/versions";
+    Assert.assertEquals(
+        "Subject must be preserved",
+        "/subjects/foo/versions/",
+        contextFilter.modifyUri(UriBuilder.fromPath(path), path, new MultivaluedHashMap<>()).getPath()
+    );
+    path = "/contexts/.ctx/subjects/foo/versions";
+    Assert.assertEquals(
+        "Subject must be preserved",
+        "/subjects/:.ctx:foo/versions/",
+        contextFilter.modifyUri(UriBuilder.fromPath(path), path, new MultivaluedHashMap<>()).getPath()
+    );
+  }
+
+  @Test
+  public void testSubjectNamedSubjects() {
+    String path = "/subjects/subjects/versions";
+    Assert.assertEquals(
+        "Subject must be preserved",
+        "/subjects/subjects/versions/",
+        contextFilter.modifyUri(UriBuilder.fromPath(path), path, new MultivaluedHashMap<>()).getPath()
+    );
+    path = "/contexts/.ctx/subjects/subjects/versions";
+    Assert.assertEquals(
+        "Subject must be preserved",
+        "/subjects/:.ctx:subjects/versions/",
+        contextFilter.modifyUri(UriBuilder.fromPath(path), path, new MultivaluedHashMap<>()).getPath()
+    );
+  }
+
+  @Test
+  public void testSubjectNamedContexts() {
+    String path = "/subjects/contexts/versions";
+    Assert.assertEquals(
+        "Subject must be preserved",
+        "/subjects/contexts/versions/",
+        contextFilter.modifyUri(UriBuilder.fromPath(path), path, new MultivaluedHashMap<>()).getPath()
+    );
+    path = "/contexts/.ctx/subjects/contexts/versions";
```

Comment:
@rayokota Does this also fix contexts with name contexts and deks with name deks?

If so, can we add unit tests for those?